### PR TITLE
Fix: Social share buttons that follow article content

### PR DESCRIPTION
**Pull request by andygout**
Fixes: [WhatsApp button at bottom of article is broken](https://jira.ft.com/browse/NFT-792)

Also means that behaviour between both sets of buttons is consistent, e.g. currently top Twitter link opens new window whereas the bottom opens new tab; fix will apply `o-share` logic to both and make bottom link also open in new window (not tab).

### CODE REVIEW

File: client/components/share/main.js [hunk 1]
```diff
@@ -4,9 +4,17 @@ import { EmailArticleData, emailArticleModes, EmailArticleView } from '@financia
...
+
+	for (let i = 0; i < shareContainers.length; i++) {
+
+		if (shareContainers[i] && !shareContainers[i].classList.contains('data-o-share--js')) {
+
+			new OShare(shareContainers[i]);
+
+		}
+
 	}
```
Comment (bjfletcher) [inline client/components/share/main.js]: aha, good find... nice, nice

another way of doing that:

```js
document.querySelectorAll('[data-o-component=o-share]')
  .filter(share => share.classList.contains('data-o-share--js'))
  .forEach(share => new OShare(share));```

either is grand :)

Comment (andygout) [inline client/components/share/main.js]: `document.querySelectorAll(...).filter is not a function`

Result of `querySelectorAll` is going to be a NodeList not an Array, so cannot call Array-exclusive methods (`filter` and `forEach`) on it.

Comment (ironsidevsquincy) [inline client/components/share/main.js]: There's a `$$` function in n-ui-foundations which calls `querySelectorAll` and returns an actual `Array` - https://github.com/Financial-Times/n-ui-foundations/blob/master/main.js#L11

Comment (andygout) [inline client/components/share/main.js]: Nice - have used that with Ben's solution here: https://github.com/Financial-Times/next-article/pull/2012

**Pull request merged**